scc/light_client/provider/rpc_test.go

```diff
@@ -0,0 +1,318 @@
+	provider := NewRpcProviderFromClient(client)
+
+	client.EXPECT().Call(gomock.Any(), "sonic_getBlockCertificates",
+		gomock.Any(), gomock.Any()).DoAndReturn(
+		func(result *[]ethapi.BlockCertificate, method string, args ...interface{}) error {
+			*result = []ethapi.BlockCertificate{
+				makeBlockCertForNumber(0),
+				makeBlockCertForNumber(1),
+			}
+			return nil
+		})
+
+	// get block certificates
+	_, err := provider.GetBlockCertificates(0, 1)
+	require.ErrorContains(err, "Too many certificates")
+}
+
+func TestRpcProvider_GetBlockCertificates_CanFetchLatestBlock(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	client := NewMockRpcClient(ctrl)
+	provider := NewRpcProviderFromClient(client)
+
+	latestBlockNumber := idx.Block(1024)
+	// block certificates
+	client.EXPECT().Call(gomock.Any(), "sonic_getBlockCertificates",
+		"latest", "0x1").DoAndReturn(
+		func(result *[]ethapi.BlockCertificate, method string, args ...interface{}) error {
+			*result = []ethapi.BlockCertificate{
+				makeBlockCertForNumber(latestBlockNumber),
+			}
+			return nil
+		})
+
+	// get block certificates
+	blockCerts, err := provider.GetBlockCertificates(latestBlock, 1)
+	require.NoError(err)
+	require.Len(blockCerts, 1)
+	require.Equal(latestBlockNumber, blockCerts[0].Subject().Number)
+}
+
+func TestRpcProvider_GetCertificates_ReturnsCertificates(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	client := NewMockRpcClient(ctrl)
+	provider := NewRpcProviderFromClient(client)
+
+	// committee certificates
+	client.EXPECT().Call(gomock.Any(), "sonic_getCommitteeCertificates",
+		gomock.Any(), gomock.Any()).DoAndReturn(
+		func(result *[]ethapi.CommitteeCertificate, method string, args ...interface{}) error {
+			*result = []ethapi.CommitteeCertificate{
+				makeCommitteeCertForPeriod(0),
+				makeCommitteeCertForPeriod(1),
+			}
+			return nil
+		})
+
+	// get committee certificates
+	comCerts, err := provider.GetCommitteeCertificates(0, 2)
+	require.NoError(err)
+	require.Len(comCerts, 2)
+	require.Equal(scc.Period(0), comCerts[0].Subject().Period)
+	require.Equal(scc.Period(1), comCerts[1].Subject().Period)
+
+	// block certificates
+	client.EXPECT().Call(gomock.Any(), "sonic_getBlockCertificates",
+		gomock.Any(), gomock.Any()).DoAndReturn(
+		func(result *[]ethapi.BlockCertificate, method string, args ...interface{}) error {
+			*result = []ethapi.BlockCertificate{
+				makeBlockCertForNumber(0),
+				makeBlockCertForNumber(1),
+			}
+			return nil
+		})
+
+	// get block certificates
+	blockCerts, err := provider.GetBlockCertificates(0, 2)
+	require.NoError(err)
+	require.Len(blockCerts, 2)
+}
+
+/// helper functions
+
+func makeCommitteeCertForPeriod(period scc.Period) ethapi.CommitteeCertificate {
+	return ethapi.CommitteeCertificate{
+		ChainId:   1,
+		Period:    uint64(period),
+		Members:   []scc.Member{},
+		Signers:   cert.BitSet[scc.MemberId]{},
+		Signature: bls.Signature{},
+	}
```

Comment:
This is practically the same as
```suggestion
	return ethapi.CommitteeCertificate{Period: uint64(period)}
```
and then it might be inlined to improve readability. Wdyt?

Reply:
https://github.com/0xsoniclabs/sonic/pull/91/commits/fef0b75b24cc8a2a8e85fa04beff49bfb96cd32f